### PR TITLE
Added Documentation for Prefix Sum of Array 

### LINKED ISSUE #245: Added Documentation for Prefix Sum of Array
## 📚 Documentation

(A clear and concise description of what the issue is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
YES.

### PR DESCRIPTION
**Pull request by RaiyanMahin**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES.

### Description

Added Doc for Prefix Sum Algorithm

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #245

### CODE REVIEW

**Comment by RaiyanMahin** [comment]
Please review , it's been 4 days
@ankitaggarwal23 @iamrajiv @HarshCasper

File: docs/competitive-programming/prefix-sum-algorithm.mdx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+thus will goes on for the rest indexes of Prefix Sum array.
+```
+
+## Algorithm
+
+ - New Prefix Sum array's first element will be same as given array's first element.
+ - Next we will run a loop from index `1` to `last` index.
+ - In Each iteration, Prefix Sum array's current element will be sum of previous element of this array and current index value of our given array.
+ - So, iterations will be like this, prefix_sum_array[1] = prefix_sum_array[0] `+` ar[1].
+ - prefix_sum_array[2] = prefix_sum_array[1] `+` ar[2] and so on.
```
Comment (vybhav72954) [inline docs/competitive-programming/prefix-sum-algorithm.mdx:30]: Put `prefix_sum_array[2]` and similar names in backticks.

Also make the algorithm less code-like and more plain text like

Comment (vybhav72954) [inline docs/competitive-programming/prefix-sum-algorithm.mdx:30]: Please put `prefix_sum_array[2]` and other similar code instances in backticks as well.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: docs/competitive-programming/prefix-sum-algorithm.mdx [hunk 2]
```diff
@@ -0,0 +1,179 @@
+---
+title: Prefix Sum Algorithm
+slug: /competitive-programming/prefix-sum-algorithm
+---
+
+## Introduction
+
```
Comment (vybhav72954) [inline docs/competitive-programming/prefix-sum-algorithm.mdx:7]: I think the Intro can be paraphrased to make it more engaging. Something like, In this article we will do ___

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: docs/competitive-programming/prefix-sum-algorithm.mdx (hunk 1 above)
Comment (RaiyanMahin) [reply docs/competitive-programming/prefix-sum-algorithm.mdx:30]: @vybhav72954  you mean to say add prefix_sum_array[3] , [4] , [5] till n?
each iteration?

**Comment by RaiyanMahin** [comment]
@vybhav72954  kindly review.

**Review by iamrajiv — APPROVED** [review]

**Comment by RaiyanMahin** [comment]
@HarshCasper Kindly review. It's been 9 days :)

File: docs/competitive-programming/prefix-sum-algorithm.mdx (hunk 1 above)
Comment (vybhav72954) [reply docs/competitive-programming/prefix-sum-algorithm.mdx:31]: `prefix_sum_array[2]` = `prefix_sum_array[1] + ar[2]` and so on.

So, iterations will be like this, `prefix_sum_array[1]` = `prefix_sum_array[0] + ar[1]`.

File: docs/competitive-programming/prefix-sum-algorithm.mdx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+In this article, we will learn about Prefix Sum Algorithm.
+Let us say, You are given an array of `N` elements. It's Prefix Sum is another array of same size.
+where PrefixSum[i] is sum of all prefix elements of the given array. Formally,
```
Comment (vybhav72954) [inline docs/competitive-programming/prefix-sum-algorithm.mdx:10]: Similarly here. The idea is that the variables, function names, operators etc should be in backticks.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
@HarshCasper Rest Looks good

**Comment by RaiyanMahin** [comment]
@vybhav72954  all changes done. Kindly review.

**Comment by RaiyanMahin** [comment]
@vybhav72954  kindly review

**Review by iamrajiv — APPROVED** [review]

**Review by vybhav72954 — APPROVED** [review]

**Comment by RaiyanMahin** [comment]
@vybhav72954 @iamrajiv kindly add labels.

**Pull request merged**